libbpf-tools/offcputime.c

```diff
@@ -314,14 +349,18 @@ int main(int argc, char **argv)
 	}
 
 	/* initialize global data (filtering options) */
-	obj->rodata->targ_tgid = env.pid;
-	obj->rodata->targ_pid = env.tid;
 	obj->rodata->user_threads_only = env.user_threads_only;
 	obj->rodata->kernel_threads_only = env.kernel_threads_only;
 	obj->rodata->state = env.state;
 	obj->rodata->min_block_ns = env.min_block_time;
 	obj->rodata->max_block_ns = env.max_block_time;
 
+	/* User space PID and TID correspond to TGID and PID in the kernel, respectively */
+	if (env.pids[0])
+		obj->rodata->filter_by_tgid = true;
+	else if (env.tids[0])
+		obj->rodata->filter_by_pid = true;
```

Comment:
Should we do both obj->rodata->filter_by_tgid and obj->rodata->filter_by_pid?

Reply:
it's modified. thank you.